package/MDAnalysis/topology/PDBParser.py

```diff
@@ -303,6 +308,28 @@ def _parseatoms(self):
         masses = guess_masses(atomtypes)
         attrs.append(Masses(masses, guessed=True))
 
+        # Guessed attributes
+        # Need to pull elements from Atom names
+        # Similar to the check for atomtypes function
+        if not any(elements):
+            warnings.warn("Element information absent.")
+            attrs.append(Elements(elements, guessed=True))
+        else:
+            elements = guess_types(elements)
+            
+            missing = np.isin(elements,'')
+            uni_elements, indices = np.unique(elements, return_inverse=True)
+            # indexlist contains the index of those elements
+            # whose atom record is missing from the column or is
+            # non-physical.
+            # This can be further used in guessing from atomnames
+            # if needed.
+            indexlist = [ j for j in range(len(indices)) if missing[j] == True]
+            warnings.warn("Element record found to be either non-physical or missing for some elements.")
+            attrs.append(Elements(elements, guessed=True))
```

Comment:
Same as above, I would simply check that all the read elements are valid, and only add them if so.